lib/galaxy/jobs/runners/__init__.py

```diff
@@ -389,6 +389,7 @@ def _find_container(
         )
 
     def _handle_runner_state(self, runner_state, job_state):
+        job_state.job_wrapper.reclaim_ownership()
```

Comment:
Is this line actually needed? Maybe add a comment about why just reclaiming in finish is not enough?

Reply:
Actually I don't know if this is needed, and I can't remember that I did any changes to `lib/galaxy/jobs/runners/__init__.py` (but it has been a while ago). It could be as well caused by the problems with git I had here: https://github.com/galaxyproject/galaxy/pull/4275

The function seems to be called only from `fail_job()` and `_finish_or_resubmit_job()` [and 1x from ./galaxy/jobs/runners/kubernetes.py]. _finish_or_resubmit_job() is called by `finish_job()` which calls `reclaim_ownership` itself (so for this line it seems redundant). 

Concluding, I don't know if its necessary, but I know that it works... any comments?

Reply:
Did some more research: I did the change (https://github.com/bernt-matthias/galaxy/commit/f842bc5e6bde81d1fb4e02432ad43fb3f52c496e) probably in response to https://github.com/galaxyproject/galaxy/issues/4308.

Reply:
https://github.com/galaxyproject/galaxy/pull/6996